src/index.d.ts

```diff
@@ -0,0 +1,268 @@
+import {
+	DataTexture,
+	RectAreaLight,
+	ShaderMaterial,
+	SpotLight,
+	Camera,
+	PerspectiveCamera,
+	MeshStandardMaterial,
+	Light,
+	Material,
+	Object3D,
+	Texture,
+	Vector2,
+	WebGLRenderer,
+	WebGLRenderTarget,
+	BufferGeometry,
+	Color,
+	ShaderMaterialParameters,
+	MeshStandardMaterialParameters,
+	Spherical,
+	Scene,
+	PMREMGenerator
+} from 'three';
+import { FullScreenQuad } from 'three/examples/jsm/postprocessing/Pass';
+import { MeshBVH, MeshBVHOptions } from 'three-mesh-bvh';
+
+// three.js type augmentation
+
+declare module 'three/src/materials/MeshStandardMaterial' {
+
+	export interface MeshStandardMaterial {
+
+		/**
+		 * Whether to render the object as completely transparent against the rest of the environment
+		 * so other objects can be composited later.
+		 *
+		 * Used by `WebGLPathTracer` from `three-gpu-pathtracer`.
+		 */
+		matte: boolean;
+		/**
+		 * Whether the object should cast a shadow.
+		 *
+		 * Used by `WebGLPathTracer` from `three-gpu-pathtracer`.
+		 */
+		castShadow: boolean;
+
+	}
+
+}
+
+// core
+
+export interface PathTracingSceneGeneratorResult {
+
+	bvhChanged: boolean;
+	bvh: MeshBVH;
+	lights: Array<Light>;
+	iesTextures: Array<DataTexture>;
+	geometry: BufferGeometry;
+	materials: Array<Material>;
+	textures: Array<Texture>;
+	objects: Array<Object3D>;
+
+}
+
+export interface BVHWorker {
+
+	generate( geometry : BufferGeometry, options?: MeshBVHOptions ) : Promise<MeshBVH>
+
+}
+
+export class PathTracingSceneGenerator {
+
+	constructor( objects?: Object3D | Array<Object3D> );
+
+	readonly initialized: boolean;
+
+	bvhOptions: MeshBVHOptions;
+	attributes: Array<string>;
+	generateBVH: boolean;
+
+	bvh: MeshBVH | null;
+	geometry: BufferGeometry;
+
+	setObjects( objects: Object3D | Array<Object3D> ): void;
+	setBVHWorker( bvhWorker: BVHWorker ): void;
+
+	generateAsync( onProgress?: ( progress: number ) => void ): Promise<PathTracingSceneGeneratorResult>;
+	generate( onProgress?: ( progress: number ) => void ): PathTracingSceneGeneratorResult;
+
+}
```

Comment:
We can leave it in for now but this is something I want to consider "deprecated", as well. I will probably remove it in another release or two. Ideally this can just be used internally by the WebGLPathTracer class and the user doesn't need to know. The deprecation warning can recommend people to use the `WebGLPathTracer` class, instead.

Reply:
Do you want to add a deprecation comment like with the old generator classes now or at some point in the future?

Reply:
I think we should add it now - I'd like to consider the `WebGLPathTracer` class the way to use the project for now.